src/socket.cr

```diff
@@ -244,13 +295,8 @@ class Socket < IO::FileDescriptor
   # ```
   def send(message)
     slice = message.to_slice
-    bytes_sent = LibC.send(fd, slice.to_unsafe.as(Void*), slice.size, 0)
-    raise Errno.new("Error sending datagram") if bytes_sent == -1
-    bytes_sent
-  ensure
-    # see IO::FileDescriptor#unbuffered_write
-    if (writers = @writers) && !writers.empty?
-      add_write_event
+    write_syscall_helper(slice, "Error sending message") do |slice|
```

Comment:
Fixed this by simply not touching the "message" parts of `Socket`, and the diff is a lot cleaner as a result.